### PR TITLE
openssh 7.6p1

### PR DESCRIPTION
**Pull request by alex**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by alex** [comment]
Two issues causing failures:

- The mirror isn't updated
- It can't load host keys for some reason; does anyone know how those worked previously?

**Comment by ilovezfs** [comment]
@alex it happens locally too:
```
iMac-TMP:~ joe$ /usr/local/sbin/sshd -D -p 8022
Could not load host key: /usr/local/etc/ssh/ssh_host_rsa_key
Could not load host key: /usr/local/etc/ssh/ssh_host_dsa_key
Could not load host key: /usr/local/etc/ssh/ssh_host_ecdsa_key
Could not load host key: /usr/local/etc/ssh/ssh_host_ed25519_key
sshd: no hostkeys available -- exiting.
```

File: Formula/openssh.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
   homepage "https://www.openssh.com/"
-  url "https://www.mirrorservice.org/pub/OpenBSD/OpenSSH/portable/openssh-7.5p1.tar.gz"
-  mirror "https://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-7.5p1.tar.gz"
-  version "7.5p1"
-  sha256 "9846e3c5fab9f0547400b4d2c017992f914222b3fd1f8eee6c7dc6bc5e59f9f0"
+  url "https://www.mirrorservice.org/pub/OpenBSD/OpenSSH/portable/openssh-7.6p1.tar.gz"
+  mirror "https://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-7.6p1.tar.gz"
+  version "7.6p1"
+  sha256 "a323caeeddfe145baaa0db16e98d784b1fbc7dd436a6bf1f479dfd5cd1d21723"
   revision 1
```
Comment (ilovezfs) [inline Formula/openssh.rb]: don't need the revision

**Comment by alex** [comment]
@ilovezfs Is that log with this revision, or 7.5p1?

**Comment by ilovezfs** [comment]
@alex prior version installed
```
iMac-TMP:homebrew-core joe$ ls /usr/local/etc/ssh
moduli				ssh_host_dsa_key.pub		ssh_host_ed25519_key.pub
org.openssh.sshd.sb		ssh_host_ecdsa_key		ssh_host_rsa_key
ssh_config			ssh_host_ecdsa_key.pub		ssh_host_rsa_key.pub
ssh_host_dsa_key		ssh_host_ed25519_key		sshd_config
```

This version installed
```
iMac-TMP:homebrew-core joe$ ls /usr/local/etc/ssh
moduli			org.openssh.sshd.sb	ssh_config		sshd_config
```
```

**Comment by ilovezfs** [comment]
> openssh: remove unnescary revision

Yes it was scary.

**Comment by alex** [comment]
I am not a good typist (and my browser didn't tell me there was a typo!)

**Comment by ilovezfs** [comment]
>I am not a good typist

Don't worry. I'm a worse typist. I promise.

**Comment by ilovezfs** [comment]
@alex it's a parallelization bug.
```
iMac-TMP:homebrew-core joe$ git diff -- Formula/openssh.rb 
diff --git a/Formula/openssh.rb b/Formula/openssh.rb
index c1d67bd..817e39b 100644
--- a/Formula/openssh.rb
+++ b/Formula/openssh.rb
@@ -48,6 +48,7 @@ class Openssh < Formula
                           "--with-pam",
                           "--with-ssl-dir=#{Formula["openssl"].opt_prefix}"
     system "make"
+    ENV.deparallelize
     system "make", "install"
 
     # This was removed by upstream with very little announcement and has
```
which should be reported.

**Comment by alex** [comment]
How did you find that? I've been gradually reading the diff and nothing had jumped out to me yet.

**Comment by ilovezfs** [comment]
@alex 
```
bash-3.2$ git diff -- Formula/openssh.rb 
diff --git a/Formula/openssh.rb b/Formula/openssh.rb
index c1d67bd..e7ae4c6 100644
--- a/Formula/openssh.rb
+++ b/Formula/openssh.rb
@@ -57,6 +57,7 @@ class Openssh < Formula
 
     buildpath.install resource("com.openssh.sshd.sb")
     (etc/"ssh").install "com.openssh.sshd.sb" => "org.openssh.sshd.sb"
+raise "here"
   end
 
   test do
bash-3.2$ rm -rf /usr/local/etc/ssh
bash-3.2$ brew install -ds openssh
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssh.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/makedepend.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssh.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/makedepend.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
==> Downloading https://www.mirrorservice.org/pub/OpenBSD/OpenSSH/portable/openssh-7.6p1.tar.gz
Already downloaded: /Users/joe/Library/Caches/Homebrew/openssh-7.6p1.tar.gz
==> Downloading https://raw.githubusercontent.com/Homebrew/patches/1860b0a74/openssh/patch-sandbox-darwin.c-apple-sandbox-named-external.diff
Already downloaded: /Users/joe/Library/Caches/Homebrew/openssh--patch-d886b98f99fd27e3157b02b5b57f3fb49f43fd33806195970d4567f12be66e71.diff
==> Downloading https://raw.githubusercontent.com/Homebrew/patches/d8b2d8c2/openssh/patch-sshd.c-apple-sandbox-named-external.diff
Already downloaded: /Users/joe/Library/Caches/Homebrew/openssh--patch-3505c58bf1e584c8af92d916fe5f3f1899a6b15cc64a00ddece1dc0874b2f78f.diff
==> Patching
==> Applying patch-sandbox-darwin.c-apple-sandbox-named-external.diff
patching file sandbox-darwin.c
Hunk #1 succeeded at 63 (offset 1 line).
==> Applying patch-sshd.c-apple-sandbox-named-external.diff
patching file sshd.c
Hunk #1 succeeded at 617 (offset -96 lines).
==> ./configure --with-libedit --with-kerberos5 --prefix=/usr/local/Cellar/openssh/7.6p1 --sysconfdir=/usr/local/etc/ssh --with-pam --with-ssl-dir=/usr/local
==> make
==> make install
==> Downloading https://opensource.apple.com/source/OpenSSH/OpenSSH-209.50.1/com.openssh.sshd.sb
Already downloaded: /Users/joe/Library/Caches/Homebrew/openssh--com.openssh.sshd.sb-7.6p1.sb
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
RuntimeError: here
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 5
When you exit this shell, you will return to the menu.
bash-3.2$ ls /usr/local/etc/ssh/
moduli			org.openssh.sshd.sb	ssh_config		sshd_config
bash-3.2$ make install
(cd openbsd-compat && /Applications/Xcode.app/Contents/Developer/usr/bin/make)
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/bin
if [ ! -d /usr/local/etc/ssh ]; then \
		./mkinstalldirs /usr/local/etc/ssh; \
	fi
/usr/local/Cellar/openssh/7.6p1/sbin/sshd -t -f /usr/local/etc/ssh/sshd_config
make[1]: Nothing to be done for `all'.
/usr/local/etc/ssh/ssh_config already exists, install will not overwrite
/usr/local/etc/ssh/sshd_config already exists, install will not overwrite
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/sbin
/usr/local/etc/ssh/moduli already exists, install will not overwrite
ssh-keygen: generating new host keys: RSA Could not load host key: /usr/local/etc/ssh/ssh_host_rsa_key
Could not load host key: /usr/local/etc/ssh/ssh_host_dsa_key
Could not load host key: /usr/local/etc/ssh/ssh_host_ecdsa_key
Could not load host key: /usr/local/etc/ssh/ssh_host_ed25519_key
sshd: no hostkeys available -- exiting.
make: [check-config] Error 1 (ignored)
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/share/man
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/share/man/man1
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/share/man/man5
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/share/man/man8
./mkinstalldirs /usr/local/Cellar/openssh/7.6p1/libexec
DSA (umask 022 ; ./mkinstalldirs /var/empty)
/usr/bin/install -c -m 0755 -s ssh /usr/local/Cellar/openssh/7.6p1/bin/ssh
ECDSA ED25519 
/usr/bin/install -c -m 0755 -s scp /usr/local/Cellar/openssh/7.6p1/bin/scp
/usr/bin/install -c -m 0755 -s ssh-add /usr/local/Cellar/openssh/7.6p1/bin/ssh-add
/usr/bin/install -c -m 0755 -s ssh-agent /usr/local/Cellar/openssh/7.6p1/bin/ssh-agent
/usr/bin/install -c -m 0755 -s ssh-keygen /usr/local/Cellar/openssh/7.6p1/bin/ssh-keygen
/usr/bin/install -c -m 0755 -s ssh-keyscan /usr/local/Cellar/openssh/7.6p1/bin/ssh-keyscan
/usr/bin/install -c -m 0755 -s sshd /usr/local/Cellar/openssh/7.6p1/sbin/sshd
/usr/bin/install -c -m 4711 -s ssh-keysign /usr/local/Cellar/openssh/7.6p1/libexec/ssh-keysign
/usr/bin/install -c -m 0755 -s ssh-pkcs11-helper /usr/local/Cellar/openssh/7.6p1/libexec/ssh-pkcs11-helper
/usr/bin/install -c -m 0755 -s sftp /usr/local/Cellar/openssh/7.6p1/bin/sftp
/usr/bin/install -c -m 0755 -s sftp-server /usr/local/Cellar/openssh/7.6p1/libexec/sftp-server
/usr/bin/install -c -m 644 ssh.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/ssh.1
/usr/bin/install -c -m 644 scp.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/scp.1
/usr/bin/install -c -m 644 ssh-add.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/ssh-add.1
/usr/bin/install -c -m 644 ssh-agent.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/ssh-agent.1
/usr/bin/install -c -m 644 ssh-keygen.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/ssh-keygen.1
/usr/bin/install -c -m 644 ssh-keyscan.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/ssh-keyscan.1
/usr/bin/install -c -m 644 moduli.5.out /usr/local/Cellar/openssh/7.6p1/share/man/man5/moduli.5
/usr/bin/install -c -m 644 sshd_config.5.out /usr/local/Cellar/openssh/7.6p1/share/man/man5/sshd_config.5
/usr/bin/install -c -m 644 ssh_config.5.out /usr/local/Cellar/openssh/7.6p1/share/man/man5/ssh_config.5
/usr/bin/install -c -m 644 sshd.8.out /usr/local/Cellar/openssh/7.6p1/share/man/man8/sshd.8
/usr/bin/install -c -m 644 sftp.1.out /usr/local/Cellar/openssh/7.6p1/share/man/man1/sftp.1
/usr/bin/install -c -m 644 sftp-server.8.out /usr/local/Cellar/openssh/7.6p1/share/man/man8/sftp-server.8
/usr/bin/install -c -m 644 ssh-keysign.8.out /usr/local/Cellar/openssh/7.6p1/share/man/man8/ssh-keysign.8
/usr/bin/install -c -m 644 ssh-pkcs11-helper.8.out /usr/local/Cellar/openssh/7.6p1/share/man/man8/ssh-pkcs11-helper.8
bash-3.2$ ls /usr/local/etc/ssh/
moduli				ssh_host_dsa_key		ssh_host_ecdsa_key.pub		ssh_host_rsa_key
org.openssh.sshd.sb		ssh_host_dsa_key.pub		ssh_host_ed25519_key		ssh_host_rsa_key.pub
ssh_config			ssh_host_ecdsa_key		ssh_host_ed25519_key.pub	sshd_config
bash-3.2$
```

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by alex** [comment]
Tests are ✅

**Comment by ilovezfs** [comment]
Thanks @alex!

**Pull request closed without merging**